### PR TITLE
INTEGRA-4189 - Replicate the LVMH Fix to Open Source AC Library

### PR DESCRIPTION
**Pull request by danbuga**

### CODE REVIEW

File: java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/AbstractAuthenticator.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import com.anaplan.client.dto.responses.RefreshTokenResp;
 import com.anaplan.client.exceptions.AnaplanAPIException;
 import com.anaplan.client.transport.ConnectionProperties;
+import feign.FeignException;
```
Comment (skharab-anaplan) [inline java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/AbstractAuthenticator.java]: unused import?

File: java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java [hunk 2]
```diff
@@ -7,23 +7,23 @@
 import com.anaplan.client.dto.OauthTokenInfo;
 import com.anaplan.client.exceptions.AnaplanAPIException;
 import com.anaplan.client.transport.ConnectionProperties;
-import org.apache.commons.lang3.StringUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
+import feign.FeignException;
```
Comment (skharab-anaplan) [inline java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java]: unused import?

File: java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java [hunk 3]
```diff
@@ -38,36 +38,24 @@ public class DeviceAuthenticator extends AbstractAuthenticator {
...
 
-  /**
-   * Initialization of the properties required for keystore generation
-   * @param connectionProperties
-   * @param authClient
-   */
+  private final DeviceAuthenticator.TokenStore store;
   public DeviceAuthenticator(ConnectionProperties connectionProperties,
-                             AnaplanAuthenticationAPI authClient) {
+      AnaplanAuthenticationAPI authClient) {
```
Comment (skharab-anaplan) [inline java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java]: nit: formatting

File: java/anaplan-connect-library/ac-http/ac-http-feign/src/main/java/com/anaplan/client/transport/client/OkHttpFeignClientProvider.java [hunk 4]
```diff
@@ -31,9 +32,10 @@ public Client createFeignClient(ConnectionProperties properties) {
       setupProxy(okHttpBuilder, properties);
     }
     // setting the read and write timeouts as well
-    okHttpBuilder.connectTimeout(properties.getHttpTimeout(), TimeUnit.SECONDS)
-        .readTimeout(properties.getHttpTimeout(), TimeUnit.SECONDS)
-        .writeTimeout(properties.getHttpTimeout(), TimeUnit.SECONDS);
+    int timeoutHTTP = Optional.ofNullable(properties.getHttpTimeout()).orElse(0);
```
Comment (skharab-anaplan) [inline java/anaplan-connect-library/ac-http/ac-http-feign/src/main/java/com/anaplan/client/transport/client/OkHttpFeignClientProvider.java:35]: should this be 0?

File: java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java [hunk 5]
```diff
@@ -83,13 +71,13 @@ public byte[] authenticate() {
   }
 
   /**
-   * Deletes the existing JKS at the existing path
+   * Clear JDK file
```
Comment (skharab-anaplan) [inline java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java]: What is JDK file?

File: java/anaplan-connect-library/ac-http/ac-http-feign/src/main/java/com/anaplan/client/transport/client/OkHttpFeignClientProvider.java (hunk 4 above)
Comment (danbuga) [reply java/anaplan-connect-library/ac-http/ac-http-feign/src/main/java/com/anaplan/client/transport/client/OkHttpFeignClientProvider.java:35]: yes

File: java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java (hunk 3 above)
Comment (danbuga) [reply java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java]: fix

File: java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java (hunk 5 above)
Comment (danbuga) [reply java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java]: jks , changed

File: java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java (hunk 2 above)
Comment (danbuga) [reply java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/DeviceAuthenticator.java]: removed

File: java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/AbstractAuthenticator.java (hunk 1 above)
Comment (danbuga) [reply java/anaplan-connect-library/ac-auth/ac-auth-default/src/main/java/com/anaplan/client/auth/AbstractAuthenticator.java]: fixed

**Review by skharab-anaplan — APPROVED** [review]
APPROVED

**Review by ajunnola-anaplan — APPROVED** [review]

**Review by magarwal-anaplan — APPROVED** [review]
Approved

**Pull request merged**